src/MahApps.Metro/Controls/NumericUpDown.cs

```diff
@@ -983,15 +983,15 @@ private void InternalSetText(double? newValue)
                 return;
             }
 
-            this.valueTextBox.Text = this.FormattedValue(newValue, this.StringFormat, this.SpecificCultureInfo);
+            this.valueTextBox.Text = this.FormattedValueString(newValue.Value, this.StringFormat, this.SpecificCultureInfo);
```

Comment:
Shouldn't we name it "PART_ValueTextBox" or something like this? I think that is a convention for WPF-Controls?

Reply:
@timunie For now I will leave it as it is. Because of further complete changes of the control itself (-> TextBox).